src/platform.js

```diff
@@ -44,16 +44,40 @@ export class Platform {
    * @return {boolean}
    */
   isSafari() {
-    return /Safari/i.test(this.win.navigator.userAgent) && !this.isChrome();
+    return /Safari/i.test(this.navigator.userAgent) && !this.isChrome();
   }
 
   /**
-   * Whether the current browser a Chrome browser.
+   * Whether the current browser is a Chrome browser.
    * @return {boolean}
    */
   isChrome() {
     // Also true for MS Edge :)
-    return /Chrome|CriOS/i.test(this.win.navigator.userAgent);
+    return /Chrome|CriOS/i.test(this.navigator.userAgent);
+  }
+
+  /**
+   * Whether the current browser is a Facebook webview.
+   * @return {boolean}
+   */
+  isFacebook() {
```

Comment:
I think it'd be more helpful if we determined if this is a twitter request, vs them having to add a bunch of extra classes? So `isTwitter = (ua, referrer) => /twitter/.test(ua) || /t\.co/.test(referrer);` adds the `amp-service-twitter` class.